djangocms_versioning/admin.py

```diff
@@ -249,12 +249,44 @@ def _get_edit_link(self, obj, request, disabled=False):
             }
         )
 
+    def _get_revert_link(self, obj, request, disabled=False):
+        """Helper function to get the html link to the revert action
+        """
+        if obj.state in (UNPUBLISHED, ARCHIVED):
+            pks_for_grouper = obj.versionable.for_grouper(
```

Comment:
Think of moving this out of per-row calculation. Currently this will run for every unpublished/archived version on the list. While the exact same code happens for published's the edit button, that doesn't happen as often (But could also be moved out of there).